tests/test_bfd_global.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/bfd_global'
+
+include Cisco
+# TestBfdGlobal - Minitest for general functionality
+# of the BfdGlobal class.
+class TestBfdGlobal < CiscoTestCase
+  @skip_unless_supported = 'bfd_global'
+  # Tests
+
+  def setup
+    super
+    config 'no feature bfd'
+  end
+
+  def teardown
+    config 'no feature bfd'
+    super
+  end
+
+  def test_create_destroy
+    # create
+    bg = BfdGlobal.new
+    assert_equal(true, Feature.bfd_enabled?)
+
+    # destroy
+    bg.destroy
+    assert_equal(bg.echo_interface, bg.default_echo_interface)
+    assert_equal(bg.fabricpath_vlan, bg.default_fabricpath_vlan) if
+      bg.fabricpath_vlan
+    assert_equal(bg.startup_timer, bg.default_startup_timer) if
+      bg.startup_timer
+  end
+
+  def test_echo_interface
+    bg = BfdGlobal.new
+    config 'interface loopback 10'
+    bg.echo_interface = 10
+    assert_equal(10, bg.echo_interface)
+    bg.echo_interface = bg.default_echo_interface
+    assert_equal(bg.default_echo_interface, bg.echo_interface)
```

Comment:
The approach we've been trying to move to is to first test for the default state, then change & test a non-default, then change & test with default:

```
  def test_echo_interface
    config 'interface loopback 10'
    bg = BfdGlobal.new
    default = bg.default_echo_interface
    assert_equal(default, bg.echo_interface)

    bg.echo_interface = 10
    assert_equal(10, bg.echo_interface)
    bg.echo_interface = default
    assert_equal(default, bg.echo_interface)
    config 'no interface loopback 10'
```